cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java

```diff
@@ -586,6 +620,27 @@ static DataprocConf create(Map<String, String> properties, @Nullable SSHPublicKe
     boolean integrityMonitoringEnabled = Boolean.parseBoolean(
       properties.getOrDefault(INTEGRITY_MONITORING_ENABLED, "false"));
 
+    boolean clusterReuseEnabled = Boolean.parseBoolean(
+      properties.getOrDefault(CLUSTER_REUSE_ENABLED, "false"));
+    int clusterReuseThresholdMinutes = getInt(properties, CLUSTER_REUSE_THRESHOLD_MINUTES,
+                                              CLUSTER_REUSE_THRESHOLD_MINUTES_DEFAULT);
+    String clusterReuseKey = null;
+    if (clusterReuseEnabled) {
+      try {
+        MessageDigest digest = MessageDigest.getInstance("SHA-1");
+        digest.update(properties.entrySet()
+                        .stream()
+                        .sorted(Map.Entry.comparingByKey())
+                        .map(e -> e.getKey() + "=" + e.getValue())
+                        .collect(Collectors.joining(","))
+                        .getBytes(StandardCharsets.UTF_8)
+        );
+        clusterReuseKey = String.format("%040x", new BigInteger(1, digest.digest()));
+      } catch (NoSuchAlgorithmException e) {
+        throw new IllegalStateException("SHA-1 algorythm is not available for cluster reuse", e);
```

Comment:
nit: algorythm -> algorithm

Reply:
When can this happen?

Reply:
1) done.
2) I think never, but it's a checked exception I need to catch.